pkgs/applications/misc/bazecor/default.nix

```diff
@@ -0,0 +1,55 @@
+{ lib
+, appimageTools
+, fetchurl
+}:
+
+appimageTools.wrapAppImage rec {
+  pname = "bazecor";
+  version = "1.3.6";
+
+  src = appimageTools.extract {
+    inherit pname version;
+    src = fetchurl {
+      url = "https://github.com/Dygmalab/Bazecor/releases/download/v.${version}/Bazecor-${version}-x64.AppImage";
+      hash = "sha256-Mz7T/AAlyfMzdfy/ZV4AEP3ClTolwr2pPzkSCPL66/w=";
+    };
+
+    # Workaround for https://github.com/Dygmalab/Bazecor/issues/370
+    postExtract = ''
+      substituteInPlace \
+        $out/usr/lib/bazecor/resources/app/.webpack/main/index.js \
+        --replace \
+          'checkUdev=()=>{try{if(c.default.existsSync(f))return c.default.readFileSync(f,"utf-8").trim()===l.trim()}catch(e){console.error(e)}return!1}' \
+          'checkUdev=()=>{return 1}'
+    '';
+  };
+
+  # also make sure to update the udev rules in ./10-dygma.rules; most recently
+  # taken from
+  # https://github.com/Dygmalab/Bazecor/blob/v1.3.6/src/main/utils/udev.ts#L6
+
+  extraPkgs = p: (appimageTools.defaultFhsEnvArgs.multiPkgs p) ++ [
+    p.glib
+  ];
+
+  # Also expose the udev rules here, so it can be used as:
+  #   services.udev.packages = [ pkgs.bazecor ];
+  # to allow non-root modifications to the keyboards.
+
+  extraInstallCommands = ''
+    mv $out/bin/bazecor-* $out/bin/bazecor
+
+    mkdir -p $out/lib/udev/rules.d
+    ln -s --target-directory=$out/lib/udev/rules.d ${./10-dygma.rules}
```

Comment:
@amesgen since this wasn't merged yet, can we add an option, so these rules are easy to override? e.g. add a `udevRules` option next to `fetchurl`.

The current udev rules don't work for the Dygma Defy (at least for me), so this would make it easier for me to reuse this derivation, otherwise I'd need to override the entire thing or just create my own like I do now.

Reply:
Would something like this work for you?
```nix
pkgs.applyPatches {
  name = "bazecor-patched-rules";
  src = pkgs.bazecor;
  postPatch = ''
    rm lib/udev/rules.d/*-10-dygma.rules
    echo 'SOMETHING ELSE' > lib/udev/rules.d/10-dygma.rules
  '';
}
```

Reply:
Sure, I wasn't aware of that function, thanks!